lib/file_store/s3_store.rb

```diff
@@ -11,10 +11,12 @@ module FileStore
   class S3Store < BaseStore
     TOMBSTONE_PREFIX ||= "tombstone/"
 
-    attr_reader :s3_helper
-
     def initialize(s3_helper = nil)
-      @s3_helper = s3_helper || S3Helper.new(s3_bucket,
+      @s3_helper = s3_helper
```

Comment:
this is NOT an aliasing problem, because all accesses have been changed to the `s3_helper` method